srtcore/sync.cpp

```diff
@@ -527,3 +535,80 @@ void srt::sync::CTimer::tick()
     m_event.notify_one();
 }
 
+
+void srt::sync::CGlobEvent::triggerEvent()
+{
+    return g_Sync.notify_one();
+}
+
+bool srt::sync::CGlobEvent::waitForEvent()
+{
+    return g_Sync.lock_wait_for(milliseconds_from(10));
+}
+
+////////////////////////////////////////////////////////////////////////////////
+//
+// CThread class
+//
+////////////////////////////////////////////////////////////////////////////////
+#ifndef USE_STDCXX_CHRONO
+
+srt::sync::CThread::CThread()
+{
+    m_thread = pthread_t();
+}
+
+srt::sync::CThread::CThread(void *(*start_routine) (void *), void *arg)
+{
+    const int st = pthread_create(&m_thread, NULL, start_routine, arg);
+    if (st != 0)
+    {
+        LOGC(mglog.Error, log << "pthread_create failed with " << st);
+        //throw std::system_error()
```

Comment:
Wait. So you actually throw or not?

Reply:
I was going to throw `std::system_error()`, but it turned out to be C++11 🙂 
So now there is a typedef
```c++
#ifdef USE_STDCXX_CHRONO
typedef std::system_error CThreadException;
#else // pthreads wrapper version
typedef ::CUDTException CThreadException;
#endif
```
and this constructor will throw the following in case of pthreads version:
```c++
throw CThreadException(CodeMajor::MJ_SYSTEMRES, CodeMinor::MN_THREAD);
```